### PR TITLE
Revert #4072 - don't use earthly for running integration tests

### PR DESCRIPTION
**Pull request by SteffenDE**
I tried to look into fixing the integration tests and could not get earthly to run locally, while it was introduced in #4072 to make local testing easier. I also don't see a real advantage, because running the tests locally in docker is also quite easy without earthly, for example with the newly added  `./docker.sh` script in the integration_tests directory.

### CODE REVIEW

**Review by josevalim — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -106,18 +106,61 @@ jobs:
...
-      - name: Download released earth
-        run: "sudo /bin/sh -c 'wget https://github.com/earthly/earthly/releases/download/v0.5.16/earthly-linux-amd64 -O /usr/local/bin/earthly && chmod +x /usr/local/bin/earthly'"
-      - name: Execute tests
-        run: earthly -P --build-arg ELIXIR=${{ matrix.elixir }} --build-arg OTP=${{ matrix.otp }} +integration-test
+      - uses: actions/checkout@v2.3.1
+      - name: Install Dependencies
+        run: |
+          mix local.rebar --force
+          mix local.hex --force
+          apk add --no-progress --update git socat make gcc libc-dev
```
Comment (mcrumm) [inline .github/workflows/ci.yml:152]:
```suggestion
          # Note if you add a package you must also add it to:
          # `integration_test/docker.sh`
          apk add --no-progress --update git socat make gcc libc-dev
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -106,18 +106,61 @@ jobs:
...
+      - name: Set up local proxies
+        run: |
+          socat TCP-LISTEN:5432,fork TCP-CONNECT:postgres:5432&
+          socat TCP-LISTEN:3306,fork TCP-CONNECT:mysql:3306&
+          socat TCP-LISTEN:1433,fork TCP-CONNECT:mssql:1433&
```
Comment (mcrumm) [inline .github/workflows/ci.yml:157]:
```suggestion
          # Note if you add a proxy you must also add it to:
          # `integration_test/docker.sh`
          socat TCP-LISTEN:5432,fork TCP-CONNECT:postgres:5432&
          socat TCP-LISTEN:3306,fork TCP-CONNECT:mysql:3306&
          socat TCP-LISTEN:1433,fork TCP-CONNECT:mssql:1433&
```

File: integration_test/docker.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# Start databases
+$COMPOSE_CMD up -d
+
+# Run test commands (adapt from .github/workflows/ci.yml if necessary)
+docker run --rm --network=integration_test_default \
+    -w $PARENT_DIR -v $PARENT_DIR:$PARENT_DIR \
+    -it hexpm/elixir:$ELIXIR-erlang-$ERLANG-$SUFFIX sh -c '
+mix local.rebar --force
+mix local.hex --force
+apk add --no-progress --update git socat make gcc libc-dev
```
Comment (mcrumm) [inline integration_test/docker.sh:34]:
```suggestion
# Note if you add a package you must also add it to:
# `.github/workflows/ci.yml`
apk add --no-progress --update git socat make gcc libc-dev
```

File: integration_test/docker.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+socat TCP-LISTEN:5432,fork TCP-CONNECT:postgres:5432&
+socat TCP-LISTEN:3306,fork TCP-CONNECT:mysql:3306&
+socat TCP-LISTEN:1433,fork TCP-CONNECT:mssql:1433&
```
Comment (mcrumm) [inline integration_test/docker.sh:37]:
```suggestion
# Note if you add a proxy you must also add it to:
# `.github/workflows/ci.yml`
socat TCP-LISTEN:5432,fork TCP-CONNECT:postgres:5432&
socat TCP-LISTEN:3306,fork TCP-CONNECT:mysql:3306&
socat TCP-LISTEN:1433,fork TCP-CONNECT:mssql:1433&
```

**Review by mcrumm — COMMENTED** [review]
Nice work! I added a few comments so hopefully we won't forget to update dependencies in both places whenever that occurs. Alternatively we could move the setup into a script?

**Comment by SteffenDE** [comment]
@mcrumm I didn't like the duplicated code either. Now it's in a separate script used in both CI and the local docker wrapper.

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**